src/poliastro/tests/tests_twobody/test_orbit.py

```diff
@@ -915,3 +916,9 @@ def test_orbit_wrong_dimensions_fails():
     with pytest.raises(ValueError) as excinfo:
         Orbit.from_vectors(Earth, bad_r, bad_v)
     assert "ValueError: Vectors must have dimension 1, got 2 and 3" in excinfo.exconly()
+
+
+def test_orbit_change_attractor_error():
+    with pytest.raises(ValueError) as excinfo:
+        iss.change_attractor(Sun)
```

Comment:
It turns out that computing the radius of the sphere of influence of the Sun makes no sense :)

> In the patched conic approximation, once an object leaves the planet's SOI, the primary/only gravitational influence is the Sun (until the object enters another body's SOI).

https://www.wikiwand.com/en/Sphere_of_influence_(astrodynamics)

I think we need two kinds of errors:

* If the attractor is set to the `Sun` and we do `set_attractor(new_attractor)`, we check the SOI of `new_attractor` and we raise the error you wrote here (`"ValueError: Body is out of new attractor's SOI."`)
* If the attractor is set to `attractor` and we do `set_attractor(attractor.parent)`, we check the SOI of `attractor` and, _if we are still inside the SOI of `attractor`_:
  - If the orbit is closed, raise `"ValueError: Body will never leave the SOI of its current attractor"`
  - If the orbit is open, just do it! (I think emitting a warning here might be too noisy, but we could as well say `"PatchedConicsWarning: Leaving the SOI of the current attractor"`

What do you think @jorgepiloto?

Reply:
I think raising these 2 errors and the warning, depending on the case, it is very useful. Hope to have this in a couple of minutes :+1: